### PR TITLE
FileTreeStudyLoader [ANT-1213]

### PR DESCRIPTION
**Pull request by JasonMarechal25**
Encapsulate Application behaviour to provide a way to read a study described with file system

### CODE REVIEW

File: src/libs/antares/file-tree-study-loader/FileTreeStudyLoader.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ *
+ * This function creates a copy of the input string_view and returns a unique_ptr to a char array.
+ * The char array is null-terminated, making it a valid C-string.
+ * Use of unique_ptr ensure proper memory management of the new raw string
+ * @param str The string_view to copy.
+ * @return std::unique_ptr<char[]> A unique_ptr to a char array that contains a copy of the input string.
+ */
+[[nodiscard]] std::unique_ptr<char[]> copy(std::string_view str) {
+    auto copy = std::unique_ptr<char[]>(new char[str.size() + 1]);
+    copy.get()[0] = '\0';
```
Comment (flomnes) [inline src/libs/antares/file-tree-study-loader/FileTreeStudyLoader.cpp]: What do you intend to do with this function ?

**Review by flomnes — COMMENTED** [review]
Some overlap with #2054 (move economy/adequacy stuff)

**Comment by JasonMarechal25** [comment]
> Some overlap with #2054 (move economy/adequacy stuff)

Yes, it's loosely based on this PR

File: src/solver/application/application.cpp [hunk 2]
```diff
@@ -189,10 +317,8 @@ void Application::prepare(int argc, char* argv[])
     // CAUTION
     // The parser contains references to members of pSettings and options,
     // don't de-allocate these.
-    auto parser = CreateParser(pSettings, options);
-    // Parse the command line arguments
 
-    handleParserReturn(parser.get());
+    parseCommandLine(options);
```
Comment (flomnes) [inline src/solver/application/application.cpp:332]: 👍🏻

File: src/solver/application/include/antares/application/application.h [hunk 3]
```diff
@@ -71,7 +71,11 @@ class Application final : public Yuni::IEventObserver<Application, Yuni::Policy:
     void resetProcessPriority() const;
 
     void writeExectutionInfo();
-private:
+    Data::Study* study() const { return pStudy.get(); }
```
Comment (flomnes) [inline src/solver/application/include/antares/application/application.h]:
```suggestion
    Data::Study& study() const { return *pStudy; }
```

File: src/libs/antares/file-tree-study-loader/FileTreeStudyLoader.cpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+ * @param argv A span of char pointers to be filled with the prepared arguments.
+ * @param study_path A string_view representing the study path.
+ * @return std::vector<std::string> A vector of strings containing the prepared arguments.
+ */
+[[nodiscard]] std::vector<std::string> prepareArgs(std::span<char*> argv, std::string_view study_path)
+{
+    using namespace std::literals::string_literals;
+    std::string arg0{""s};
+    std::string arg1{study_path};
+    std::string arg2{"-s"s};
```
Comment (flomnes) [inline src/libs/antares/file-tree-study-loader/FileTreeStudyLoader.cpp]:
```
  -s, --named-mps-problems   Export named constraints and variables in mps
                             (both optim).
```
So we force the export of named MPS problems ?

File: src/libs/antares/file-tree-study-loader/include/antares/file-tree-study-loader/FileTreeStudyLoader.h [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#pragma once
+#include "antares/study-loader/IStudyLoader.h"
+#include "antares/study/load-options.h"
+#include "antares/solver/misc/options.h"
+#include <antares/writer/i_writer.h>
```
Comment (flomnes) [inline src/libs/antares/file-tree-study-loader/include/antares/file-tree-study-loader/FileTreeStudyLoader.h]:
```suggestion
```

File: src/libs/antares/file-tree-study-loader/include/antares/file-tree-study-loader/FileTreeStudyLoader.h [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * @brief Loads a study from the file tree.
+     *
+     * This function prepares the arguments required by the Antares Solver application and then
+     * calls the application's prepare method.
+     * It then returns the study loaded by the application.
+     *
+     * @return std::shared_ptr<Antares::Data::Study> A shared_ptr to the loaded Study object.
+     */
+    [[nodiscard]] std::unique_ptr<Antares::Data::Study> load() const override;
```
Comment (flomnes) [inline src/libs/antares/file-tree-study-loader/include/antares/file-tree-study-loader/FileTreeStudyLoader.h:57]: fwd-declare `Antares::Data::Study` ?

File: src/libs/antares/study-loader/include/antares/study-loader/IStudyLoader.h [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#pragma once
+#include <antares/study/study.h>
```
Comment (flomnes) [inline src/libs/antares/study-loader/include/antares/study-loader/IStudyLoader.h]: Use fwd-declaration instead of inclusion ?

**Comment by flomnes** [comment]
Merge conflicts

File: src/libs/antares/file-tree-study-loader/FileTreeStudyLoader.cpp (hunk 4 above)
Comment (JasonMarechal25) [reply src/libs/antares/file-tree-study-loader/FileTreeStudyLoader.cpp]: Indeed it should not be necessary

Comment (JasonMarechal25) [reply src/libs/antares/file-tree-study-loader/FileTreeStudyLoader.cpp]: Such options will have to be handled by options.ini or expansion of the API

File: src/solver/application/include/antares/application/application.h (hunk 3 above)
Comment (JasonMarechal25) [reply src/solver/application/include/antares/application/application.h]: Removed. Function was unused

**Review by flomnes — APPROVED** [review]

**Pull request merged**